ios/engine/KMEI/KeymanEngine/Classes/KeymanInputViewController.swift

```diff
@@ -0,0 +1,308 @@
+      textDocumentProxy.insertText(text)
+    }
+  }
+
+  @objc public func updatedFragment(_ fragment: String) {
+    if fragment.isEmpty {
+      return
+    }
+
+    if fragment.contains("insertText") {
+      processInsertText(fragment)
+    } else if fragment.contains("menuKeyUp") {
+      if Manager.shared.isKeyboardMenuVisible {
+        return
+      }
+
+      switch globeKeyTapBehaviour {
+      case .switchToNextKeyboard:
+        if let nextIndex = Manager.shared.switchToNextKeyboard(), nextIndex <= 0 {
+          advanceToNextInputMode()
+        }
+      case .switchToNextInputMethod:
+        advanceToNextInputMode()
+      case .doNothing:
+        break
+      }
+    } else if fragment.contains("showKeyboardMenu") {
+      switch menuBehaviour {
+      case .showAlways,
+           .showIfMultipleKeyboards where keyboardListCount > 1:
+        Manager.shared.showKeyboardMenu(self, closeButtonTitle: menuCloseButtonTitle)
+      case .showIfMultipleKeyboards, // keyboardListCount() <= 1
+           .showNever:
+        break
+      }
+    } else if fragment.contains("hideKeyboard") {
+      dismissKeyboard()
+    }
+  }
+
+  private func setConstraints() {
+    let topBarHeight = isTopBarEnabled ? KeymanInputViewController.topBarHeight : 0
+    let viewsDict = ["bar": topBarImageView!, "container": containerView!]
+    barHeightConstraints = NSLayoutConstraint.constraints(
+      withVisualFormat: "V:[bar(\(topBarHeight))]", metrics: nil, views: viewsDict)
+
+    let screenWidth = UIScreen.main.bounds.width
+    let barWidthConstraints = NSLayoutConstraint.constraints(
+      withVisualFormat: "H:[bar(>=\(Int(screenWidth)))]", metrics: nil, views: viewsDict)
+    let barVerticalPositionConstraints = NSLayoutConstraint.constraints(
+      withVisualFormat: "V:|-0-[bar]", metrics: nil, views: viewsDict)
+    let barHorizontalPositionConstraints = NSLayoutConstraint.constraints(
+      withVisualFormat: "H:|-0-[bar]", metrics: nil, views: viewsDict)
+
+    view.addConstraints(barHeightConstraints)
+    view.addConstraints(barWidthConstraints)
+    view.addConstraints(barVerticalPositionConstraints)
+    view.addConstraints(barHorizontalPositionConstraints)
+
+    // note this appears to relate to the rotation fix bug; may not work on iPads as 1000 is not enough
+    containerHeightConstraints = NSLayoutConstraint.constraints(
+      withVisualFormat: "V:[container(\(Int(kmInputView.frame.height) % 1000))]",
+      metrics: nil, views: viewsDict)
+
+    let containerWidthConstraints = NSLayoutConstraint.constraints(
+      withVisualFormat: "H:[container(>=\(UInt(screenWidth)))]", metrics: nil, views: viewsDict)
+    let containerVericalPositionConstraints = NSLayoutConstraint.constraints(
+      withVisualFormat: "V:[container]-0-|", metrics: nil, views: viewsDict)
+    let containerHorizontalPositionConstraints = NSLayoutConstraint.constraints(
+      withVisualFormat: "H:|-0-[container]", metrics: nil, views: viewsDict)
```

Comment:
What happened to `"H:|[container]-0-"`?  Granted, I'm generalizing based on the pattern here in comparison to the Obj-C code, but the original Obj-C method has two extra constraints in comparison to the Swift method, and both correspond to 'H' and '-0-' after the identifier within a string.

I assume their omission is intentional?  How does that work?

Reply:
That format was used to set `cView_constraint_POS_HR`, which was never used so it got axed.